src/analysis/ExpressionExplorer.jl

```diff
@@ -112,6 +112,14 @@ Base.push!(x::Set) = x
 # HELPER FUNCTIONS
 ###
 
+function explore_new_scope(ex::Expr, scopestate::ScopeState)::SymbolsState
+    # Because we are entering a new scope, we create a copy of the current scope state, and run it through the expressions.
+    innerscopestate = deepcopy(scopestate)
+    innerscopestate.inglobalscope = false
+
+    return explore!(Expr(:block, ex), innerscopestate)
+end
```

Comment:
This pattern is used in lots of other places. Maybe you could also replace those with a call to `explore_inner_scoped`?

Reply:
I replaced it for `let`, `for` and `while` :+1: However, I could not use it on the [function declaration](https://github.com/fonsp/Pluto.jl/blob/master/src/analysis/ExpressionExplorer.jl#L386-L391) where it seemed different enough.